src/file-path.cpp

```diff
@@ -8,8 +8,31 @@
 #include <quick-lint-js/utf-16.h>
 
 #if QLJS_HAVE_WINDOWS_H
-#include <Windows.h>
-#include <pathcch.h>
+//#include <Windows.h>
+// The following was copy-pasted from Windows.h. Without an architecture defined, winnt.h would bark incessantly
+// ------
+#if !defined(_68K_) && !defined(_MPPC_) && !defined(_X86_) && !defined(_IA64_) && !defined(_AMD64_) && !defined(_ARM_) && !defined(_ARM64_) && defined(_M_IX86)
+#define _X86_
+#if !defined(_CHPE_X86_ARM64_) && defined(_M_HYBRID)
+#define _CHPE_X86_ARM64_
+#endif
+#endif
+
+#if !defined(_68K_) && !defined(_MPPC_) && !defined(_X86_) && !defined(_IA64_) && !defined(_AMD64_) && !defined(_ARM_) && !defined(_ARM64_) && defined(_M_AMD64)
+#define _AMD64_
+#endif
+
+#if !defined(_68K_) && !defined(_MPPC_) && !defined(_X86_) && !defined(_IA64_) && !defined(_AMD64_) && !defined(_ARM_) && !defined(_ARM64_) && defined(_M_ARM)
+#define _ARM_
+#endif
+
+#if !defined(_68K_) && !defined(_MPPC_) && !defined(_X86_) && !defined(_IA64_) && !defined(_AMD64_) && !defined(_ARM_) && !defined(_ARM64_) && defined(_M_ARM64)
+#define _ARM64_
+#endif
+// ------
```

Comment:
Must fix: Do not include Microsoft-copyright unlicensed code in our repository.

Reply:
`Windows.h` seems to be unavoidable here.